ftp.go

```diff
@@ -387,7 +402,12 @@ func (c *ServerConn) openDataConn() (net.Conn, error) {
 		return nil, err
 	}
 
-	return c.options.dialer.Dial("tcp", net.JoinHostPort(host, strconv.Itoa(port)))
+	addr := net.JoinHostPort(host, strconv.Itoa(port))
+	if c.options.dialFunc != nil {
+		return c.options.dialFunc("tcp", addr)
+	} else {
+		return c.options.dialer.Dial("tcp", addr)
```

Comment:
can you move this return stateout out of the else block as suggested by golint ?